### PR TITLE
Support for `plan/2` and `process_results/3` funs for coverage fsm

### PR DESCRIPTION
**Pull request by russelldb**
Enables 2i pagination.

See https://github.com/basho/riak_kv/pull/535 for more details

### CODE REVIEW

File: src/riak_core_coverage_fsm.erl [hunk 1]
```diff
@@ -272,3 +282,31 @@ terminate(Reason, _StateName, _State) ->
...
+                    Mod:process_results(VNode, Results, ModState) end;
+        false ->
+            fun(_VNode, Results, ModState) ->
+                    Mod:process_results(Results, ModState) end
+    end.
+
+exports(Function, Exports) ->
+    proplists:is_defined(Function, Exports).
+
+exports_arity(Function, Arity, Exports) ->
```
Comment (engelsanchez) [inline src/riak_core_coverage_fsm.erl:308]: Probably simpler to just do `lists:member(Arity, proplists:get_all_values(Function, Exports))`

File: src/riak_core_coverage_fsm.erl [hunk 2]
```diff
@@ -112,7 +112,9 @@ behaviour_info(_) ->
                 required_responses :: pos_integer(),
                 response_count=0 :: non_neg_integer(),
                 timeout :: timeout(),
-                vnode_master :: atom()
+                vnode_master :: atom(),
+                plan_fun :: function(),
```
Comment (engelsanchez) [inline src/riak_core_coverage_fsm.erl:116]: I learned something new here. I can't see function() in the list of built-in types http://www.erlang.org/doc/reference_manual/typespec.html, but it's obviously used in the OTP code and ours.

**Comment by russelldb** [comment]
Addressed comment, it was a much better idea, I did it, thanks!

**Comment by engelsanchez** [comment]
I'm done here. Tested along the other 2i pagination changes, looking good.

:+1: :dancer: :boat:

**Pull request merged**